android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -5,47 +5,363 @@
+                    String decodedUrl = decodeBase64SvgUrl(url);
+                    validUrl = sanitizeSvg(decodedUrl);
+                } else if (isUtf8(url)) {
+                    validUrl = sanitizeSvg(url);
+                } else {
+                    // Unsupported URL type.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+                if (TextUtils.isEmpty(validUrl)) {
+                    // This may happen for invalid, or corrupted URLs.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+            } else {
+                validUrl = url;
+            }
+
+            WebContentsFactory webContentsFactory = new WebContentsFactory();
+            WebContents webContents =
+                    webContentsFactory.createWebContentsWithWarmRenderer(profile, false);
+            webContents.downloadImage(new GURL(validUrl), // Url
+                    true, // isFavIcon - Does not send or accept cookies when `true`.
+                    WalletConstants.MAX_BITMAP_SIZE_FOR_DOWNLOAD, // maxBitmapSize
+                    false, // bypassCache
+                    (id, httpStatusCode, imageUrl, bitmaps, originalImageSizes) -> { // callback
+                        ImageFetcherFacade imageFetcherFacade;
+                        Iterator<Bitmap> iterBitmap = bitmaps.iterator();
+                        Iterator<Rect> iterSize = originalImageSizes.iterator();
+                        Bitmap bestBitmap = null;
+                        Rect bestSize = new Rect(0, 0, 0, 0);
+                        while (iterBitmap.hasNext() && iterSize.hasNext()) {
+                            Bitmap bitmap = iterBitmap.next();
+                            Rect size = iterSize.next();
+                            if (size.width() > bestSize.width()
+                                    && size.height() > bestSize.height()) {
+                                bestBitmap = bitmap;
+                                bestSize = size;
+                            }
+                        }
+                        if (bestSize.width() == 0 || bestSize.height() == 0) {
+                            imageFetcherFacade = null;
+                        } else {
+                            BitmapDrawable bitmapDrawable = new BitmapDrawable(bestBitmap);
+                            imageFetcherFacade = new ImageFetcherFacade(bitmapDrawable);
+                        }
+                        loadImage(imageFetcherFacade, context, isCircular, imageView, callback);
+                    });
+        } else {
+            ImageFetcher imageFetcher = ImageFetcherFactory.createImageFetcher(
+                    ImageFetcherConfig.NETWORK_ONLY, profile.getProfileKey());
+            if (isGif(url)) {
+                imageFetcher.fetchGif(
+                        Params.create(new GURL(url), UNUSED_CLIENT_NAME), gifImage -> {
```

Comment:
Passed a simple string "unused" as a client name, are there better options?

Reply:
I wouldn't worry about it, just pass what it accepts, all reporting is disabled.